libs/kml/serdes_geojson.cpp

```diff
@@ -0,0 +1,224 @@
+          umapOptions != propsJson.end() && umapOptions->second.is_object())
+      {
+        JsonTMap const umap_options = umapOptions->second.get_object();
+        // Parse color from properties['_umap_options']['color']
+        if (auto const color = getStringFromJsonMap(umap_options, "color"))
+        {
+          auto colorRGBA = ParseHexOsmGarminColor(*color);
+          if (colorRGBA)
+            bookmark.m_color = ColorData{.m_rgba = *colorRGBA};
+        }
+
+        // TODO: Store '_umap_options' as a JSON string in some bookmark field.
+      }
+
+      bookmark.m_point = mercator::FromLatLon(latitude, longitude);
+      m_fileData.m_bookmarksData.push_back(bookmark);
+    }
+    else if (auto const * unknownGeometry = std::get_if<GeoJsonGeometryUnknown>(&feature.geometry))
+    {
+      LOG(LWARNING, ("GeoJson contains unsupported geometry type:", unknownGeometry->type));
+    }
+  }
+
+  // Copy tracks from parsed geoJsonData into m_fileData.
+  for (auto & feature : geoJsonData.features)
+  {
+    if (auto const * lineGeometry = std::get_if<GeoJsonGeometryLine>(&feature.geometry))
+    {
+      std::vector<std::vector<double>> lineCoords = lineGeometry->coordinates;
+
+      // Convert GeoJson properties to KML properties
+      auto const & props_json = feature.properties;
+      TrackData track;
+
+      // Parse "name" or "label"
+      if (auto const name = getStringFromJsonMap(props_json, "name"))
+        kml::SetDefaultStr(track.m_name, *name);
+      else if (auto const label = getStringFromJsonMap(props_json, "label"))
+        kml::SetDefaultStr(track.m_name, *label);
+
+      // Parse color
+      std::unique_ptr<ColorData> lineColor;
+      if (auto const stroke = getStringFromJsonMap(props_json, "stroke"))
+      {
+        auto const colorRGBA = ParseHexOsmGarminColor(*stroke);
+        if (colorRGBA)
+          lineColor = std::make_unique<ColorData>(PredefinedColor::None, *colorRGBA);
+      }
+
+      // UMap custom properties
+      if (auto const umapOptions = props_json.find("_umap_options");
+          umapOptions != props_json.end() && umapOptions->second.is_object())
+      {
+        JsonTMap umap_options = umapOptions->second.get_object();
+        // Parse color from properties['_umap_options']['color']
+        if (auto const color = getStringFromJsonMap(umap_options, "color"))
+        {
+          auto colorRGBA = ParseHexOsmGarminColor(*color);
+          if (colorRGBA)
+            lineColor = std::make_unique<ColorData>(PredefinedColor::None, *colorRGBA);
+        }
+
+        // TODO: Store '_umap_options' as a JSON string in some bookmark field.
+      }
+
+      if (lineColor)
+        track.m_layers.push_back(TrackLayer{.m_color = *lineColor});
+
+      // Copy coordinates
+      std::vector<geometry::PointWithAltitude> points;
+      points.resize(lineCoords.size());
+      for (size_t i = 0; i < lineCoords.size(); i++)
+      {
+        auto pointCoords = lineCoords[i];
```

Comment:
Avoid unnecessary copy here.
```suggestion
        auto const & pointCoords = lineCoords[i];
```